### PR TITLE
Validate Credential Names during Testing

### PR DESCRIPTION
**Pull request by ianlivingstone**
Grafton now validates whether or not the credential names it's receiving
back are valid. They must be valid names in accordance with Shell and
Utilities volume of IEEE Std 1003.1-2001.

A function is also exported for use by client libraries.

### CODE REVIEW

File: credentials.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"regexp"
+)
+
+// NameRegexp is the regular expression for determining if a credential name is
+// valid or not.
+//
+// Specified in Shell and Utilities volume of IEEE 1003.1-2001.
+var NameRegexp = regexp.MustCompile("^[A-Z][A-Z0-9_]{0,127}$")
```
Comment (jbowes) [inline credentials.go]: IMO exporting just the func is enough.

File: credentials_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.in, func(t *testing.T) {
+			gm.RegisterTestingT(t)
+
+			gm.Expect(ValidCredentialName(tc.in)).To(gm.Equal(tc.out))
+		})
+	}
+}
```
Comment (jbowes) [inline credentials_test.go]: 👍

File: credentials.go (hunk 1 above)
Comment (jbowes) [inline credentials.go]: you could export the uncompiled string form of the regexp as a const, and then use it in acceptance/credentials.go's output msg.

**Comment by ianlivingstone** [comment]
This is actually broken - I'm going to try pinning our bootstrap dependencies

**Pull request merged**